yt_dlp/extractor/bilibili.py

```diff
@@ -996,3 +996,115 @@ def _real_extract(self, url):
             self._entries(series_id), series_id, series_info.get('title'), series_info.get('description'),
             categories=traverse_obj(series_info, ('styles', ..., 'title'), expected_type=str_or_none),
             thumbnail=url_or_none(series_info.get('horizontal_cover')), view_count=parse_count(series_info.get('view')))
+
+
+class BiliLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://live.bilibili.com/(?P<room_id>\d+)'
+
+    _TESTS = [
+        {
+            'url': 'https://live.bilibili.com/196',
+            'info_dict': {
+                'id': '196',
+                'ext': 'flv',
+            },
+            'params': {
+                'skip_download': True,
+            },
+            'skip': 'extractor won\'t work while stream is not live'
+        }, {
+            'url': 'https://live.bilibili.com/196?broadcast_type=0&is_room_feed=1?spm_id_from=333.999.space_home.strengthen_live_card.click',
+            'only_matching': True
+        }
+    ]
+
+    def _call_api_base(self, path, video_id):
+        json = self._download_json('https://api.live.bilibili.com' + path, video_id)
+        if json.get('code') != 0:
+            if json.get('message'):
+                raise ExtractorError(json.get('message'))
+            else:
+                raise ExtractorError('Unable to download JSON metadata')
+        return json.get('data')
+
+    def _call_api_info(self, room_id):
+        json = self._call_api_base(f'/room/v1/Room/get_info?id={room_id}', room_id)
+        if json.get('live_status') == 0:
+            raise ExtractorError('Streamer is not live')
+        return json
+
+    def _call_api_stream(self, room_id, qn):
+        json = self._call_api_base(f'/xlive/web-room/v2/index/getRoomPlayInfo?room_id={room_id}&no_playurl=0&mask=0&qn={qn}&platform=web&protocol=0,1&format=0,2&codec=0,1',
+                                   room_id)
+        return json
+
+    def _real_extract(self, url):
+        room_id = self._match_valid_url(url).group('room_id')
```

Comment:
```suggestion
        room_id = self._match_id(url)
```
rename the group accordingly